session_rpcserver.go

```diff
@@ -152,6 +160,20 @@ func (s *sessionRpcServer) AddSession(_ context.Context,
 				continue
 			}
 
+			// If the action specified was equal to the
+			// readOnlyAction keyword, then this is taken to mean
+			// that the permissions for all read-only URIs should be
+			// granted.
+			if op.Action == readOnlyAction {
+				readPerms := s.cfg.permMgr.ActivePermissions(
+					true,
+				)
+
+				permissions = append(permissions, readPerms...)
```

Comment:
Do we do a de-duplication somewhere? Because maybe the `***readonly***` overlaps with URIs matched by a regex?

Reply:
(added in the last commit so that the first 2 commits dont need to be re-reviewed. Hope that that is ok)